src/Sarif/Core/Invocation.cs

```diff
@@ -4,20 +4,40 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 using System.Reflection;
 
 namespace Microsoft.CodeAnalysis.Sarif
 {
     public partial class Invocation
     {
-        public static Invocation Create(bool emitMachineEnvironment = false)
+        private IEnumerable<string> PropertiesToLog {get; set; }
+
+        public static Invocation Create(
+            bool emitMachineEnvironment = false,
+            IEnumerable<string> propertiesToLog = null)
         {
-            var invocation = new Invocation();
+            var invocation = new Invocation
+            {
+                PropertiesToLog = propertiesToLog?.Select(p => p.ToUpperInvariant()).ToList()
+            };
 
             invocation.StartTime = DateTime.UtcNow;
-            invocation.ProcessId = System.Diagnostics.Process.GetCurrentProcess().Id;
-            invocation.WorkingDirectory = Environment.CurrentDirectory;
-            invocation.CommandLine = Environment.CommandLine;
+
+            if (invocation.ShouldLog(nameof(ProcessId)))
+            {
+                invocation.ProcessId = System.Diagnostics.Process.GetCurrentProcess().Id;
+            }
+
+            if (invocation.ShouldLog(nameof(WorkingDirectory)))
+            {
+                invocation.WorkingDirectory = Environment.CurrentDirectory;
+            }
+
+            if (invocation.ShouldLog(nameof(CommandLine)))
+            {
+                invocation.CommandLine = Environment.CommandLine;
+            }
 
             if (emitMachineEnvironment)
```

Comment:
hm, should emitting of machine environment be handled by the property name control instead as a simplification?